pkg/chartverifier/verifier.go

```diff
@@ -77,64 +74,33 @@ func (c *verifier) subConfig(name string) *viper.Viper {
 	}
 }
 
-type version struct{}
-
-func (ver *version) getVersion(debug bool) (string, error) {
-
-	procExec := exec.NewProcessExecutor(debug)
-	oc := tool.NewOc(procExec)
-
-	// oc.GetVersion() returns an error both in case the oc command can't be executed and
-	// the value for the OpenShift version key not present.
-	return oc.GetVersion()
-}
-
 func (c *verifier) Verify(uri string) (*Report, error) {
 
 	chrt, _, err := checks.LoadChartFromURI(uri)
 	if err != nil {
 		return nil, err
 	}
 
-	// oc.GetVersion() returns an error both in case the oc command can't be executed and
-	// the value for the OpenShift version key not present.
-	osVersion, getVersionErr := c.version.getVersion(c.settings.Debug)
-
-	// From this point on, an error is set and osVersion is empty.
-	if getVersionErr != nil && c.openshiftVersion != "" {
-		osVersion = c.openshiftVersion
-	}
-
-	// osVersion is empty only if an error happened and a default value
-	// informed by the user hasn't been informed.
-	if osVersion == "" {
-		return nil, OpenShiftVersionErr(getVersionErr.Error())
-	}
-
-	// osVersion is guaranteed to have a value, not yet validated as a
-	// semver value.
-	if _, err := semver.NewVersion(osVersion); err != nil {
-		return nil, OpenShiftSemVerErr(err.Error())
-	}
-	// osVersion is guaranteed to a valid value from here onwards.
-
 	result := NewReportBuilder().
 		SetToolVersion(c.toolVersion).
 		SetChartUri(uri).
-		SetChart(chrt).
-		SetCertifiedOpenShiftVersion(osVersion)
+		SetChart(chrt)
 
 	for _, name := range c.requiredChecks {
 		check, ok := c.registry.Get(name)
 		if !ok {
 			return nil, CheckNotFoundErr(name)
 		}
 
+		holder := AnnotationHolder{Holder: result,
+			CertifiedOpenShiftVersionFlag: c.openshiftVersion}
+
 		r, checkErr := check.Func(&checks.CheckOptions{
-			HelmEnvSettings: c.settings,
-			URI:             uri,
-			Values:          c.values,
-			ViperConfig:     c.subConfig(name),
+			HelmEnvSettings:  c.settings,
+			URI:              uri,
+			Values:           c.values,
+			ViperConfig:      c.subConfig(name),
+			AnnotationHolder: &holder,
```

Comment:
Another option to adding `AnnotationHolder` as a member of `CheckOptions` is to change the `check.Func` type to get a hold of the aforementioned `map[string]string` bag.

Reply:
Also not keen on this approach having to check for a value and then have a mapping of the key used to the annotation. Once again I think we need to figure a way to do this but I am not seeing a great solution just yet.